state-proofs/sui_state_proof/sources/sui_state_proof.move

```diff
@@ -0,0 +1,37 @@
+module sui_state_proof::sui_state_proof {
+    use std::vector;
+    use sui::object::{Self, ID, UID};
+    use sui::event;
+    use sui::tx_context::TxContext;
+
+
+    struct DWalletCap has key, store {
+        id: UID,
+    }
+
+    struct DWalletNetworkRequest has copy, drop {
+        cap_id: ID,
+        message: vector<u8>,
+    }
+
+    public fun create_cap(ctx: &mut TxContext): DWalletCap {
+        let cap = DWalletCap {
+            id: object::new(ctx),
+        };
+
+        event::emit(DWalletNetworkRequest {
```

Comment:
It is a very ackward design choice you made here, choosing to use the same event for two functionally independent events: creation and message approval. These should be seperated, as well as the native code that checks for them. It was also confusing in the dwallet side's move code.